### PR TITLE
drivers/watchdog/stm3u5: Specify max of iwdg prescaler for stm32u5

### PR DESCRIPTION
**Pull request by nfaurant**
In file drivers/watchdog/wdt_iwdg_stm32.c set prescaler up to 1024.

### CODE REVIEW

**Review by katsuster — DISMISSED** [review]

**Review by erwango — CHANGES_REQUESTED** [review]
Looks fine, but please update comment around `IWDG_SR_UPDATE_TIMEOUT` definition as it explicitly refers to 256 as highest prescaler value.

**Comment by FRASTM** [comment]
LGTM, with Erwan's comment

**Comment by nfaurant** [comment]
> LGTM, with Erwan's comment

The comment is updated

File: drivers/watchdog/wdt_iwdg_stm32.c [hunk 1]
```diff
@@ -39,7 +43,7 @@
 
 /*
  * Status register needs 5 LSI clock cycles divided by prescaler to be updated.
- * With highest prescaler (256) and considering clock variation, we will wait
+ * With highest prescaler (256 or 1024 for STM32U5) and considering clock variation, we will wait
```
Comment (martinjaeger) [inline drivers/watchdog/wdt_iwdg_stm32.c:46]: You will need to add a line break, otherwise it will not pass CI (max. 80 characters per line are allowed).

**Review by martinjaeger — CHANGES_REQUESTED** [review]
I guess @erwango was mainly referring to the Kconfig comments and limits which need an update:
https://github.com/zephyrproject-rtos/zephyr/blob/ca0702283fba0949539c355192b1231c32917c6b/drivers/watchdog/Kconfig.stm32#L18-L32

Can you please update the `Kconfig.stm32` as well?

**Comment by erwango** [comment]
> I guess @erwango was mainly referring to the Kconfig comments and limits which need an update:

Well I missed it actually. But that's required indeed. Thanks @martinjaeger

File: drivers/watchdog/wdt_iwdg_stm32.c [hunk 2]
```diff
@@ -19,7 +19,11 @@
 #include "wdt_iwdg_stm32.h"
 
 #define IWDG_PRESCALER_MIN	(4U)
+#if defined(CONFIG_SOC_SERIES_STM32U5X)
+#define IWDG_PRESCALER_MAX	(1024U)
```
Comment (FRASTM) [inline drivers/watchdog/wdt_iwdg_stm32.c:23]: maybe with LL_IWDG_PRESCALER_256 and  LL_IWDG_PRESCALER_1024  definition from LL in a 
#ifdef LL_IWDG_PRESCALER_1024 
...

**Comment by erwango** [comment]
@nfaurant Are you planning to update this PR ?

**Comment by erwango** [comment]
@nfaurant Do you plan to update this PR?

**Comment by erwango** [comment]
See https://github.com/zephyrproject-rtos/zephyr/pull/52050

**Pull request closed without merging**